src/main/java/org/opentripplanner/middleware/docs/VersionControlledSwaggerUpdater.java

```diff
@@ -0,0 +1,41 @@
+package org.opentripplanner.middleware.docs;
+
+import org.opentripplanner.middleware.OtpMiddlewareMain;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+import static org.opentripplanner.middleware.docs.PublicApiDocGenerator.getSwaggerDocsAsString;
+
+/**
+ * This class is solely intended to be executed on a development machine
+ * to update the version-controlled spark-swagger output.
+ */
+public class VersionControlledSwaggerUpdater {
+    public static final String SWAGGER_CONTROLLED_FILE = "src/main/resources/spark-swagger-controlled-output.yaml";
+
+    private static final Logger LOG = LoggerFactory.getLogger(VersionControlledSwaggerUpdater.class);
+
+    public static void main(String[] args) throws IOException, InterruptedException {
+        OtpMiddlewareMain.main(new String[0]);
+        updateSwaggerDocs();
```

Comment:
This is getting closer, but is it necessary to start OtpMiddlewareMain here? Can the docs be retrieved directly from the `PublicApiDocGenerator` class and written to a file without starting up a server?

Reply:
The server has to be run to accomplish this task. We use the swagger generated at runtime as skeleton to create the public-facing version of it, at runtime too.

Reply:
Very well, I'll approve then.

Reply:
Unfortunately, `initializeHttpEndpoints` is not enough. There are persistence classes involved which must be also initialized, so better run `OtpMiddleware.main` as in the tests.

Reply:
I added a comment about this in https://github.com/ibi-group/otp-middleware/pull/59/commits/b327a799b261096bd67fcaf0d680d92b1090fc2a